### PR TITLE
optimize glob.match for common cases

### PR DESCRIPTION
**Pull request by jrieken**
Assuming many, many glob patterns look like `**/*.txt` or `{**/*.js,**/*.ts}` it's IMO fair to optimise those `string#endsWith` checks

### CODE REVIEW

**Comment by jrieken** [comment]
related to #5438 and #5539

File: src/vs/base/common/glob.ts [hunk 1]
```diff
@@ -264,6 +268,16 @@ export function match(arg1: string | IExpression, path: string, siblings?: strin
 
 	// Glob with String
 	if (typeof arg1 === 'string') {
+
+		if (trivia1.test(arg1)) {
+			// common pattern: **/*.txt just need endsWith check
+			return strings.endsWith(path, arg1.substr(4)); // '**/*'.length === 4
```
Comment (outcoldman) [inline src/vs/base/common/glob.ts]: you are assuming that most of these patterns have 3 letters in file type, but we know that it is not always the case, like `.js`, `.ts`. Better will be to actually get the match

```
const trivia1 = /^\*\*\/\*(\.\w+)$/;

if ((let t1match = trivia1.match(arg1)) != null) {
    return strings.endsWith(path, t2match[1]);
}
```

Comment (jrieken) [inline src/vs/base/common/glob.ts]: > you are assuming that most of these patterns have 3 letters in file type, 

please explain

Comment (outcoldman) [inline src/vs/base/common/glob.ts]: Plus current implementation has an issue, for example if I will have two file types `foo.myext` and `bar.notmyext` and have a pattern `**/*.myext` - both files will match this extension.

Please add unit tests for this improvement.

Comment (outcoldman) [inline src/vs/base/common/glob.ts]: @jrieken actually never mind. You are stripping first 4 letters from `arg1`, not the last 4.

Comment (jrieken) [inline src/vs/base/common/glob.ts]: @outcoldman the tests you were looking for: https://github.com/Microsoft/vscode/blob/master/src/vs/base/test/common/glob.test.ts

**Pull request merged**